### PR TITLE
Show an alert on the home page after an unauthorized login attempt

### LINKED ISSUE #801: Unauthorized screen for someone trying to log into Federalist is unfriendly
The screen for a user who can't log into Federalist is an ugly UNAUTHORIZED message in plain text. We should make it friendlier!

### PR DESCRIPTION
**Pull request by jseppi**
Ref #801

Also add simple test case for the `Home` component with and without the unauthorized alert showing.

It looks like this:

screen shot 2017-05-30 at 5 53 40 pm

To Do:
- [x] Adjust authentication tests to reflect the new behavior
- [x] Review/adjust alert message (@wslack)

### CODE REVIEW

**Comment by jmhooper** [comment]
It looks like the auth request specs are failing. It is expecting a 401, but we changed it so instead it is redirecting.

**Comment by brandocalrissian** [comment]
will check it out!

**Comment by wslack** [comment]
Hold on this

**Comment by wslack** [comment]
edited! Thoughts?

File: frontend/components/home.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Home extends React.Component {
           <div className="usa-alert usa-alert-error usa-alert-home" role="alert">
             <div className="usa-alert-body">
               <h3 className="usa-alert-heading">Unauthorized</h3>
-              <p className="usa-alert-text">Sorry, you are not authorized to log in to Federalist.</p>
+              <p className="usa-alert-text">Apologies; you don't have access to Federalist! Please contact the Federalist team if this is in error.</p>
```
Comment (jseppi) [inline frontend/components/home.js]: I think you'll have to escape the `'` in `don't` --> `don\'t`

Or, even better, use `&apos;` --> `don&apos;t`

Comment (wslack) [inline frontend/components/home.js]: done!

**Comment by jseppi** [comment]
Looks good to me, @wslack. I updated the screenshot in the root comment.

**Comment by wslack** [comment]
I'm happy but travis is not.....

**Comment by jseppi** [comment]
Linting errors since the PR that added linting on modified source files is now in staging branch. I'll fix those.

**Comment by jseppi** [comment]
bleh, I think the associated Travis build got stuck during GitHub's API outage this morning...

**Comment by wslack** [comment]
we can reboot with another small commit right?

**Comment by jseppi** [comment]
Or close and reopen, it turns out.

File: .eslintrc [hunk 2]
```diff
@@ -1,4 +1,15 @@
 {
   "extends": "airbnb",
-  "parser": "babel-eslint",
+  "rules": {
+    "react/jsx-filename-extension": [0],
```
Comment (jseppi) [inline .eslintrc]: This would require touching a bunch of stuff, so I have disabled it for now

File: .eslintrc [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+    "comma-dangle": ["error",
+      {
+        "arrays": "always-multiline",
+        "objects": "always-multiline",
+        "imports": "always-multiline",
+        "exports": "always-multiline",
+        "functions": "never"
```
Comment (jseppi) [inline .eslintrc]: Same for this one, which is having dangling commas in function argument lists.

File: test/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "env": {
+    "mocha": true
+  },
+  "rules": {
+      "no-unused-expressions": [0] // because chai's expect is weird
```
Comment (jseppi) [inline test/.eslintrc]: This is because a lot of chai expectations, like `expect(...).to.be.equal;` use getter methods instead of functions :\

**Comment by wslack** [comment]
Ok, so this is now ready to go into `staging`?

Also James, why use redirects instead of a 401 response, as Brandon mentioned?

**Comment by jseppi** [comment]
I did this implementation initially as my first pairing session with @jmhooper, mostly as a learning exercise.

I actually agree now that a `401` response and a custom error page would be better, but it will be more of an implementation lift than what is here.

I think how it could better be done is:
1. extract common site components (header, footer, etc) into a template
1. refactor the home page to use the template, and stop loading its content with React
1. refactor the single-page app to use the template
1. create a custom 401 page that uses the template and return that after an unauthorized user authenticates with GitHub

**Comment by brandocalrissian** [comment]
I looked at the changes since the last review and still approve. Merging.

**Pull request merged**